crates/uv-cli/src/lib.rs

```diff
@@ -2932,6 +2932,16 @@ pub struct RunArgs {
     /// By default, environment modifications are omitted, but enabled under `--verbose`.
     #[arg(long, env = EnvVars::UV_SHOW_RESOLUTION, value_parser = clap::builder::BoolishValueParser::new(), hide = true)]
     pub show_resolution: bool,
+
+    /// Number of times this process has been recursively invoked by uv
+    /// run. Used to detect and error on potential infinite recursion.
+    #[arg(long, hide = true, env = EnvVars::UV_RUN_RECURSION_DEPTH)]
+    pub recursion_depth: Option<u32>,
+
+    /// Number of times that uv run will recursively invoke itself before
+    /// giving up.
+    #[arg(long, hide = true, env = EnvVars::UV_RUN_MAX_RECURSION_DEPTH)]
+    pub max_recursion_depth: Option<u32>,
```

Comment:
I can see a case for adding this to the CLI but... we could also just read it directly. Since you already did this, it seems okay to leave it (as a hidden option).

Reply:
Yeah, I made this configurable primarily to support testing, but this seemed like something that a user could theoretically want to control, and it isn't too hard to plumb this to where it needs to go.